admin_webapp/controllers/users.py

```diff
@@ -0,0 +1,176 @@
+
+
+# TODO: this is broken because of a faulty TapirUser-Demographic relationship
+def suspect_listing(per_page:int, page: int) -> dict:
+    session = get_db(current_app).session
+    report_stmt = (select(TapirUsers)
+                   .options(joinedload(TapirUsers.demographics))
+                   .filter(Demographics.flag_suspect == "1")
+                   .limit(per_page).offset((page -1) * per_page))
+
+    count_stmt = (select(func.count(TapirUsers.user_id))
+                  .where(Demographics.flag_suspect == "1"))
+
+    users = session.scalars(report_stmt)
+    count = session.execute(count_stmt).scalar_one()
+    pagination = Pagination(query=None, page=page, per_page=per_page, total=count, items=None)
+    return dict(pagination=pagination, count=count, users=users)
+
+
+def moderator_listing() -> dict:
+    session = get_db(current_app).session
+
+    count_stmt = select(func.count(func.distinct(Moderators.user_id)))
+
+    # mods = session.scalars(report_stmt)
+    mods = (
+    session.query(
+        Moderators.user_id,
+        # TODO: Note, is checking for empty string in subject_class enough?
+        func.group_concat(func.concat(Moderators.archive, case([(Moderators.subject_class != '', '.',)], else_=''), Moderators.subject_class), order_by=(Moderators.archive, Moderators.subject_class), separator=', ').label('archive_subject_list'),
+        TapirUsers,
+    )
+    .join(TapirUsers, Moderators.user_id == TapirUsers.user_id)
+    # .join(TapirNicknames, Moderators.user_id == TapirNicknames.user_id)
+
+    .group_by(Moderators.user_id)
+    .order_by(TapirUsers.last_name) # order by nickname?
+    .all()
+    )
+    count = session.execute(count_stmt).scalar_one()
+
+    return dict(count=count, mods=mods)
+
+# TODO: optimize this once a relationship between TapirUsers and Moderators model is established
+def moderator_by_category_listing() -> dict:
+    session = get_db(current_app).session
+
+    count_stmt = select(func.count(func.distinct(Moderators.user_id)))
+
+    mods = (
+    session.query(
+        Moderators.user_id,
+        # TODO: Note, is checking for empty string in subject_class enough?
+        func.group_concat(func.concat(Moderators.archive, case([(Moderators.subject_class != '', '.',)], else_=''), Moderators.subject_class), order_by=(Moderators.archive, Moderators.subject_class), separator=', ').label('archive_subject_list'),
+        TapirUsers
+    )
+    .join(TapirUsers, Moderators.user_id == TapirUsers.user_id)
+    .group_by(Moderators.user_id)
+    .all()
+    )
+    count = session.execute(count_stmt).scalar_one()
+    mods_map = defaultdict(list)
+    mods = sorted(mods, key=lambda x: x[1])
+    for mod in mods:
+        for pair in mod.archive_subject_list.split(','):
+            user = session.scalar((select(TapirUsers)
+            .where(
+                TapirUsers.user_id == mod.user_id
+            )))
+            mods_map[pair].append(user)
+
```

Comment:
Also may need to paginate this. We don't think of mods for a category as getting very large but result sets often creeps up over time when the system starts to get used in ways we didn't originally think of.

You can start with a large page size.